mypy/test/testpackages.py

```diff
@@ -0,0 +1,118 @@
+from contextlib import contextmanager
+import os
+import site
+import sys
+from typing import Generator
+from unittest import TestCase, main
+
+import mypy.api
+from mypy.build import get_package_dirs
+from mypy.test.config import package_path
+from mypy.test.helpers import run_command
+from mypy.util import try_find_python2_interpreter
+
+
+SIMPLE_PROGRAM = """
+from typedpkg.sample import ex
+a = ex([''])
+reveal_type(a)
+"""
+
+
+class TestPackages(TestCase):
+
+    @contextmanager
+    def install_package(self, pkg: str,
+                        python: str = sys.executable) -> Generator[None, None, None]:
+        """Context manager to temporarily install a package from test-data/packages/pkg/"""
+        working_dir = os.path.join(package_path, pkg)
+        install_cmd = [python, '-m', 'pip', 'install', '.']
+        # if we aren't in a virtualenv, install in the
+        # user package directory so we don't need sudo
+        if not hasattr(sys, 'real_prefix') or python != sys.executable:
+            install_cmd.append('--user')
+        returncode, lines = run_command(install_cmd, cwd=working_dir)
+        if returncode != 0:
+            self.fail('\n'.join(lines))
+        try:
+            yield
+        except AssertionError as e:
+            package_dirs = get_package_dirs(python)
```

Comment:
I added this for my own testing purposes since I originally couldn't repro the Appveyor failures. I can remove it if its too "noisy".